pkgs/development/tools/misc/cquery/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub, makeWrapper
+, cmake, llvmPackages, ncurses }:
+
+let
+  src = fetchFromGitHub {
+    owner = "cquery-project";
+    repo = "cquery";
+    rev = "e45a9ebbb6d8bfaf8bf1a3135b6faa910afea37e";
+    sha256 = "049gkqbamq4r2nz9yjcwq369zrmwrikzbhfza2x2vndqzaavq5yg";
+    fetchSubmodules = true;
+  };
+
+  stdenv = llvmPackages.stdenv;
+
+in
+stdenv.mkDerivation rec {
+  name    = "cquery-${version}";
+  version = "2018-03-25";
+
+  inherit src;
+
+  nativeBuildInputs = [ cmake makeWrapper ];
+  buildInputs = with llvmPackages; [ clang clang-unwrapped llvm ncurses ];
+
+  cmakeFlags = [
+    "-DSYSTEM_CLANG=ON"
+    "-DCLANG_CXX=ON"
+  ];
+
+  postFixup = ''
+    # We need to tell cquery where to find the standard library headers.
+
+    args="\"-isystem\", \"${if (stdenv.hostPlatform.libc == "glibc") then stdenv.cc.libc.dev else stdenv.cc.libc}/include\""
+    args+=", \"-isystem\", \"${llvmPackages.libcxx}/include/c++/v1\""
```

Comment:
if `llvmPackages.libcxx` is in `buildInputs, `$NIX_CXXSTDLIB_COMPILE` should be used instead.

Reply:
I tried $NIX_CXXSTDLIB_COMPILE at first, but that starts with the paths to libstdc++ if we don't force libcxxStdenv. In my tests libclang seems to have a few problems with libstdc++, it works flawlessly with libc++.

Reply:
ok, then.